### PR TITLE
Export Cloudflare runtime types

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Fixes https://github.com/withastro/astro/issues/5222

## Testing

Tested with a project

## Docs

N/A

### CODE REVIEW

File: packages/integrations/cloudflare/runtime.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export type {
```
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/integrations/cloudflare/runtime.d.ts:1]: We do emit types to dist. But TypeScript doesn't read the `exports` field (in certain versions of TypeScript) and so it has to find them at the root like this.

[the remark this answers is not in the document]
Comment (delucis) [reply packages/integrations/cloudflare/runtime.d.ts:1]: Ahhh, right — would it make sense to update the `build` script to `astro-scripts build \"src/**/*.ts\" && tsc && cp dist/runtime.d.ts runtime.d.ts` and `.gitignore` the one in root?

File: packages/integrations/cloudflare/runtime.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  WorkerRuntime,
+  PagesRuntime
+} from './dist/runtime';
+
+export {
+  getRuntime
+} from './dist/runtime';
```
Comment (bluwy) [inline packages/integrations/cloudflare/runtime.d.ts:8]:
```suggestion
} from './dist/runtime.js';

export {
  getRuntime
} from './dist/runtime.js';
```

I believe these should end with `.js` for node16 resolution support. We also have the same pattern for the main Astro types.

Comment (matthewp) [reply packages/integrations/cloudflare/runtime.d.ts:8]: Doesn't look like we do: https://github.com/withastro/astro/blob/main/packages/astro/types.d.ts#L1-L2

File: packages/integrations/cloudflare/runtime.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthewp) [reply packages/integrations/cloudflare/runtime.d.ts:1]: I think this approach is simpler than depending on a build that could be changed accidentally.

File: packages/integrations/cloudflare/runtime.d.ts (hunk 2 above)
Comment (bluwy) [reply packages/integrations/cloudflare/runtime.d.ts:8]: Ah looks like we're only doing it for ambient types (<https://github.com/withastro/astro/blob/0e163533ea8a9800b9e288fd0ae6853ace878101/packages/astro/astro-jsx.d.ts#L26-L32>). Maybe it's fine to not have it for the module files, looks like it's also working today.

**Review by bluwy — APPROVED** [review]

File: packages/integrations/cloudflare/runtime.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (delucis) [reply packages/integrations/cloudflare/runtime.d.ts:1]: I thought the manual version might be prone to becoming out of date if we change `src` and not this file, but if the runtime file won’t change very often maybe that’s fine 👍

[the remark this answers is not in the document]
Comment (matthewp) [reply packages/integrations/cloudflare/runtime.d.ts:1]: Yeah, that is a risk. We are doing this the same way in the main astro file. Not sure if there's a good solution.

**Pull request merged**